File: Haskell/ABC/ABC001/ABC001_A_v02.hs
-- [contest]  ABC 001
-- [task]     A
-- [URL]      https://atcoder.jp/contests/abc001/tasks/abc001_1
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc001/submissions/6401389 ; AC

-- modules to import --

import qualified Data.ByteString.Char8
import qualified Data.Maybe
import           Prelude

-- functions for this task --

read_Integer = Prelude.fst . Data.Maybe.fromJust . Data.ByteString.Char8.readInteger
get_Integer  = read_Integer <$> Data.ByteString.Char8.getLine

-- the main process is as follows --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given data
        height_1st <- get_Integer
        height_2nd <- get_Integer

        -- STEP.02
        -- calculate & output the answer of this task
        Prelude.print $ Prelude.uncurry (-) (height_1st, height_2nd)

-- EOF --

File: CPP14(GCC 5.4.1)/ABC/ABC001/ABC001_A_v02.cpp
// [contest]  ABC 001
// [task]     A
// [URL]      https://atcoder.jp/contests/abc001/tasks/abc001_1
// [compiler] C++ 14 (GCC 5.4.1)
// [status]   https://atcoder.jp/contests/abc001/submissions/9090135 : WA
//            https://atcoder.jp/contests/abc001/submissions/9090167 : AC

#include <iostream>

namespace abc001
{
	namespace task_A
	{
		struct SnowDepth
		{
			public: int fst;
			public: int snd;

			public: int change ()
			{
				return this->fst - this->snd;
			}
		};
		
	} // namespace task_A
	
} // namespace abc001


int main (void)
{
	// variables for main process
	abc001::task_A::SnowDepth given_data;

	// STEP.01
	// read out the height of the snow
	std::cin >> given_data.fst >> given_data.snd;

	// STEP.02
	// output the result
	std::cout << given_data.change() << std::endl;

	// STEP.END
	return EXIT_SUCCESS;
}

/* EOF */
